project/assets/scripts/component/component-constructor.js

```diff
@@ -1,24 +1,24 @@
 const ComponentConstructor = function (options) {
-    if (options.login) {
+    if (options === 'login') {
         return {
             view: new LoginPage(),
-            setEvents: new LoginEvents(),
+            events: new LoginEvents(),
```

Comment:
Would we be able to hide event details from the outside world? Could we encapsulate the event setup inside a module initialization routine?